normalizer_test.go

```diff
@@ -0,0 +1,583 @@
+			expected: "SELECT ?",
+			statementMetadata: StatementMetadata{
+				Tables:   []string{},
+				Comments: []string{},
+				Commands: []string{},
+			},
+		},
+		{
+			input:    "SELECT * FROM users WHERE id = ?",
+			expected: "SELECT * FROM users WHERE id = ?",
+			statementMetadata: StatementMetadata{
+				Tables:   []string{},
+				Comments: []string{},
+				Commands: []string{},
+			},
+		},
+		{
+			input:    "SELECT id as ID, name as Name FROM users WHERE id IN (?, ?)",
+			expected: "SELECT id AS ID, name AS Name FROM users WHERE id IN ( ? )",
+			statementMetadata: StatementMetadata{
+				Tables:   []string{},
+				Comments: []string{},
+				Commands: []string{},
+			},
+		},
+		{
+			input:    `TRUNCATE TABLE datadog`,
+			expected: "TRUNCATE TABLE datadog",
+			statementMetadata: StatementMetadata{
+				Tables:   []string{},
+				Comments: []string{},
+				Commands: []string{},
+			},
+		},
+	}
+
+	normalizer := NewNormalizer(
+		WithCollectComments(false),
+		WithCollectCommands(false),
+		WithCollectTables(false),
+		WithKeepSQLAlias(true),
+	)
+
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			got, statementMetadata, err := normalizer.Normalize(test.input)
+			if err != nil {
+				t.Errorf("error during normalization: %v", err)
+			}
+			if got != test.expected {
+				t.Errorf("got %q, expected %q", got, test.expected)
+			}
+			assert.Equal(t, statementMetadata, &test.statementMetadata)
+		})
+	}
+}
+
+func TestNormalizerFormatting(t *testing.T) {
+	tests := []struct {
+		queries  []string
+		expected string
+	}{
+		{
+			queries: []string{
+				"SELECT id,name, address FROM users where id = ?",
+				"select id, name, address FROM users where id = ?",
+				"select id as ID, name as Name, address FROM users where id = ?",
+			},
+			expected: "SELECT id, name, address FROM users WHERE id = ?",
+		},
+		{
+			queries: []string{
+				"SELECT id,name, address FROM users where id IN (?, ?,?, ?)",
+				"select id, name, address FROM users where id IN ( ? )",
+				"select id, name, address FROM users where id IN ( ? )",
+				"select id, name, address FROM users where id IN (?,?,?)",
+			},
+			expected: "SELECT id, name, address FROM users WHERE id IN ( ? )",
+		},
+		{
+			queries: []string{
+				"SELECT * FROM discount where description LIKE ?",
+				"select * from discount where description LIKE ?",
+				"select * from discount where description like ?",
+			},
+			expected: "SELECT * FROM discount WHERE description LIKE ?",
+		},
+	}
+
+	normalizer := NewNormalizer(
+		WithCollectComments(false),
+	)
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			for _, query := range test.queries {
+				got, _, err := normalizer.Normalize(query)
+				if err != nil {
+					t.Errorf("error during normalization: %v", err)
+				}
+				if got != test.expected {
+					t.Errorf("got %q, expected %q", got, test.expected)
+				}
+			}
+		})
+	}
+}
+
+func TestNormalizerRepeatedExecution(t *testing.T) {
```

Comment:
@alexandre-normand This test should guarantee that normalizer always produce identical output

Reply:
Nice. Thanks for writing this one. ❤️